src/main/java/seedu/address/model/Model.java

```diff
@@ -84,4 +85,26 @@ public interface Model {
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredContactList(Predicate<Contact> predicate);
+
+    /**
+     * Returns true if a team with the same identity as {@code team} exists in CodeConnect.
+     */
+    boolean hasTeam(Team team);
+    /**
+     * Adds the given team.
+     * {@code team} must not already exist in CodeConnect.
+     */
+    void addTeam(Team team);
+    /**
+     * Deletes the given team.
+     * The team must exist in CodeConnect.
+     */
+    void deleteTeam(Team target);
+    /**
+     * Replaces the given team {@code target} with {@code editedTeam}.
+     * {@code target} must exist in CodeConnect.
+     * The contact identity of {@code editedTeam} must not be the same as another existing team in CodeConnect.
+     */
+    void setTeam(Team target, Team editedTeam);
```

Comment:
Is it more appropriate to change it from ```setTeam``` to ```replaceTeam``` ?

Reply:
I was following the same naming for the `*Contact()` methods, as in `setContact()`. I think it was named that way so that the method names are consistent, even though it doesn't really match the method's behavior (see `ModelManager`). Also, if we were to change the name, I think `updateTeam()` reflects the intended use of the method better.